zeppelin-server/src/main/java/org/apache/zeppelin/rest/NotebookRestApi.java

```diff
@@ -334,7 +335,31 @@ public Response createNote(String message) throws IOException {
     LOG.info("Create new note by JSON {}", message);
     NewNoteRequest request = NewNoteRequest.fromJson(message);
     AuthenticationInfo subject = new AuthenticationInfo(user);
-    Note note = notebook.createNote(subject);
+    Note note;
+
+    //set default interpreter
+    if (request != null && request.getInterpreter() != null) {
+      String interpreter = request.getInterpreter();
+      InterpreterSettingManager manager = notebook.getInterpreterSettingManager();
+      List<String> interpreterIDs = manager.getDefaultInterpreterSettingList();
+      String defaultId = null;
+      for (String id : interpreterIDs) {
+        if (manager.get(id).getName().equals(interpreter)) {
+          defaultId = id;
+          break;
+        }
+      }
+      if (defaultId != null) {
+        interpreterIDs.remove(defaultId);
+        interpreterIDs.add(0, defaultId);
```

Comment:
sorry, I didnt get your idea. Do you mean interpreterID duplicate test should be added? Why?Remove then add , no duplicate.